datafusion/core/src/config.rs

```diff
@@ -161,6 +163,14 @@ impl BuiltInConfigs {
     }
 }
 
+fn scalar_from_string(value: String, target_type: &DataType) -> Result<ScalarValue> {
+    let value = ScalarValue::Utf8(Some(value));
+    let cast_options = cast::CastOptions { safe: false };
```

Comment:
This is clever -- it may be worth putting directly in `ScalarValue` perhaps, perhaps something like

```rust
impl ScalarValue {
  // ..
  fn try_new_string(value: String, target_type: &DataType) -> Result<Self> {
    ...
  }
}
```

Reply:
Added `try_from_string` to `ScalarValue` to match `try_from_array` 👍